tools/scoring/score/dimensions/entity_identification.py

```diff
@@ -32,24 +32,27 @@ class EntityIdentification(Dimension):
   # rather than `translations` to do its calculations
   category = DimensionCategories.COMPLEX
 
-  def _list_ids_reporting(self, file: DeserializedFile) -> List[CloudDeviceId]:
+  def _list_ids_reporting(self, *, file: DeserializedFile,
+                          exclude_noncanonical: bool) -> List[CloudDeviceId]:
+    """Generates list of `cloud_device_id`s representing reporting entities."""
+    reporting_entities = filter(self.is_entity_reporting, file.values())
+    filtered = filter(
+        self.is_entity_canonical,
+        reporting_entities) if exclude_noncanonical else reporting_entities
+    return [entity.cloud_device_id for entity in filtered]
+
+  def _list_ids_virtual(self, *, file: DeserializedFile,
+                        exclude_noncanonical: bool) -> List[CloudDeviceId]:
     """Generates list of `cloud_device_id`s representing
-    reporting entities with canonical types."""
+    reporting entities that are linked to by virtual entities."""
+    virtual_entities = filter(self.is_entity_virtual, file.values())
+    filtered = filter(
+        self.is_entity_canonical,
+        virtual_entities) if exclude_noncanonical else virtual_entities
     return [
-        entity.cloud_device_id
-        for entity in filter(self.is_entity_canonical,
-                             filter(self.is_entity_reporting, file.values()))
-    ]
-
-  def _list_ids_virtual(self, file: DeserializedFile) -> List[CloudDeviceId]:
-    """Generates list of `cloud_device_id`s representing
-    reporting entities with canonical types that
-    are linked to by virtual entities."""
-    return [
-        cloud_device_id for source_list in (
-            (file[link.source].cloud_device_id for link in entity.links)
-            for entity in filter(self.is_entity_canonical,
-                                 filter(self.is_entity_virtual, file.values())))
+        cloud_device_id for source_list in ((file[link.source].cloud_device_id
```

Comment:
simplify the syntax here too with an `old` for loop please

Reply:
https://github.com/google/digitalbuildings/pull/561/commits/9eec8f21541d609a54f52c3bb3207c4fd4aa5832

Reply:
https://github.com/google/digitalbuildings/pull/552/commits/b26745428813dce292c368603a7b003ba22318f9